File: app-mvc-java/src/main/java/com/naveed/samples/apputils/events/SimpleEvent.java
package com.naveed.samples.apputils.events;

/**
 * Created by naveedali on 9/9/17.
 */

public class SimpleEvent implements BaseEvent{

    public int id;
    public String message;
    public boolean status;

    public SimpleEvent(boolean pStatus, String pMessage) {
        this.status = pStatus;
        this.message = pMessage;

    }

    public SimpleEvent(boolean pStatus, int pId, String pMessage) {
        this.status = pStatus;
        this.id = pId;
        this.message = pMessage;

    }


    @Override
    public int getEventId() {
        return id;
    }

    @Override
    public boolean getStatus() {
        return status;
    }

    @Override
    public String getMessage() {
        return message;
    }
}

File: app-mvp/src/main/java/com/naveed/samples/apputils/events/SimpleEvent.kt
package com.naveed.samples.apputils.events

/**
 * Created by naveedali on 9/9/17.
 */

class SimpleEvent : BaseEvent {

    override var eventId: Int = 0
    override var message: String
    override var status: Boolean = false

    constructor(pStatus: Boolean, pMessage: String) {
        this.status = pStatus
        this.message = pMessage

    }

    constructor(pStatus: Boolean, pId: Int, pMessage: String) {
        this.status = pStatus
        this.eventId = pId
        this.message = pMessage

    }
}
